### ISSUE TITLE
`basic_units.py` doesn't display negative radians correctly

### ISSUE THREAD

**Issue by Harnesser** [matplotlib/matplotlib#11531]
### Bug report

**Bug summary**

Changing the example `radian_ticks.py` to plot a waveform from a negative to positive radians range displays incorrect tick labels.

```python
    x = [val*radians for val in np.arange(-7, 8, 0.01)]
```

![original_neg_radians](https://user-images.githubusercontent.com/490167/42054948-a9ecc522-7b0d-11e8-8031-8c48d0023e53.png)

The issues:
1. "0" is repeated
2. "-1pi/2" should just be "-pi/2"
3. The negative labels are just wrong

**Fix**
Change `rad_fn()` in `basic_units.py`:

```python
# radians formatting
def rad_fn(x, pos=None):
    if x >= 0:
        n = int((x / np.pi) * 2.0 + 0.25)
    else:
        n = int((x / np.pi) * 2.0 - 0.25)
        
    if n == 0:
        return '0'
    elif n == 1:
        return r'$\pi/2$'
    elif n == 2:
        return r'$\pi$'
    elif n == -1:
        return r'$-\pi/2$'
    elif n == -2:
        return r'$-\pi$'
    elif n % 2 == 0:
        return r'$%s\pi$' % (n//2,)
    else:
        return r'$%s\pi/2$' % (n,)
```

![fixed](https://user-images.githubusercontent.com/490167/42055151-33c8c9e4-7b0e-11e8-80e8-a712d4b5f6e6.png)

**Comment by tacaswell** [comment]
@Harnesser Thanks for reporting this.  Can you open a PR with that change?

**Comment by Harnesser** [comment]
Ok, let's try this (#11532)

**Comment by anntzer** [comment]
Closed by #11532.
